src/services/cherry-picker.ts

```diff
@@ -156,6 +161,26 @@ export class CherryPicker {
     return node
   }
 
+  // Fetch app/node's service log from redis
+  async fetchRawServiceLogs(blockchain: string, rawNodeIDs: string[]): Promise<{ [id: string]: string }> {
+    const redisKeys = [] as string[]
+    const rawServiceLogs = {} as { [id: string]: string }
+
+    rawNodeIDs.forEach(async (rawNodeID) => {
+      redisKeys.push(blockchain + '-' + rawNodeID + '-service')
+    })
+
+    const rawRedisLogs = await this.redis.mget(redisKeys)
```

Comment:
`mget` in the end is still O(N) so I wouldn't say they're completely reduced but, definitely doing it in one single call will be faster due to doing a single trip.

Reply:
Yea true I'm assuming the problem is the network traffic and not the actual calls on redis